support/azureutil/azureutil.go

```diff
@@ -217,3 +226,64 @@ func VerifyResourceGroupLocationsMatch(ctx context.Context, hc *hyperv1.HostedCl
 
 	return nil
 }
+
+// GetAzureCredentialsFromSecret gets the Service Principal client ID, client secret, and tenant ID from the credentials
+// secret. This function will be modified a bit once the Microsoft sidecar containers support Managed Identity are
+// delivered (expected Oct 2024).
+func GetAzureCredentialsFromSecret(ctx context.Context, c client.Client, namespace, credsName string) (*corev1.Secret, error) {
+	var azureCredentials corev1.Secret
+
+	// Retrieve the Azure credentials secret to extract the needed fields for the managed identity containers
+	credentialsSecretName := client.ObjectKey{Namespace: namespace, Name: credsName}
+	if err := c.Get(ctx, credentialsSecretName, &azureCredentials); err != nil {
+		return nil, fmt.Errorf("failed to get secret %s: %w", credentialsSecretName, err)
+	}
+
+	for _, expectedKey := range []string{"AZURE_CLIENT_ID", "AZURE_CLIENT_SECRET", "AZURE_TENANT_ID"} {
```

Comment:
Should we expect the subscription_id as well or do we retrieve that from elsewhere ?

Reply:
No that is directly exposed in the API - https://github.com/openshift/hypershift/blob/da39c6232f098798325bc41eecaad1fe0c2f0090/api/hypershift/v1beta1/hostedcluster_types.go#L1851